### PR TITLE
Add plugin socket related debug docs.

### PR DESCRIPTION
**Pull request by anusha-ragunathan**
Signed-off-by: Anusha Ragunathan <anusha.ragunathan@docker.com>

### CODE REVIEW

File: docs/extend/index.md [hunk 1]
```diff
@@ -289,3 +289,20 @@ follows:
 $ docker-runc exec -t f52a3df433b9aceee436eaada0752f5797aab1de47e5485f1690a073b860ff62 sh
 ```
 
+#### Using curl to debug plugin socket issues.
+
+To verify if the plugin API socket that the docker daemon communicates with is responsive, use curl. In this example, we will make API calls from the docker host to volume and network plugins using curl 7.47.0 to ensure that the plugin is listening on the said socket. For a well functioning plugin, then these basic requests should work. Note that plugin sockets are available on the host under `/var/run/docker/plugins/<pluginID>`
+
+
+```bash
+curl -H "Content-Type: application/json" -XPOST -d '{}' --unix-socket /var/run/docker/plugins/e8a37ba56fc879c991f7d7921901723c64df6b42b87e6a0b055771ecf8477a6d/plugin.sock http:/VolumeDriver.List
```
Comment (thaJeztah) [inline docs/extend/index.md]: ugh, yes cURL 7.4 is broken, and only works with invalid URL's here. cURL 7.5 requires this to be;

http://foobar/VolumeDriver.List

Comment (anusha-ragunathan) [inline docs/extend/index.md]: Do you think its good to provide examples for both versions?

File: docs/extend/index.md [hunk 2]
```diff
@@ -289,3 +289,20 @@ follows:
 $ docker-runc exec -t f52a3df433b9aceee436eaada0752f5797aab1de47e5485f1690a073b860ff62 sh
 ```
 
+#### Using curl to debug plugin socket issues.
+
+To verify if the plugin API socket that the docker daemon communicates with is responsive, use curl. In this example, we will make API calls from the docker host to volume and network plugins using curl 7.47.0 to ensure that the plugin is listening on the said socket. For a well functioning plugin, then these basic requests should work. Note that plugin sockets are available on the host under `/var/run/docker/plugins/<pluginID>`
```
Comment (thaJeztah) [inline docs/extend/index.md]: Started to try a different wording, but maybe someone of the docs team has a better suggestion;

```Markdown
You can verify if a plugin's API socket is functioning correctly using `cURL`.
The following example is executed on the docker host that the plugin is installed on,
and uses `cURL` to connect with the plugin socket to performs an API request. The
location of the plugin's socket is based on its ID, and can be found at `/var/run/docker/plugins/<pluginID>`.
```

File: docs/extend/index.md (hunk 1 above)
Comment (anusha-ragunathan) [inline docs/extend/index.md]: Added a note.

File: docs/extend/index.md (hunk 2 above)
Comment (anusha-ragunathan) [inline docs/extend/index.md]: I'm not sure how this wording is any better than the existing one :) Unless we have a strong reason to change what's existing, I'd like to keep it as-is.

File: docs/extend/index.md [hunk 3]
```diff
@@ -289,3 +289,20 @@ follows:
...
+
+{"Mountpoint":"","Err":"","Volumes":[{"Name":"myvol1","Mountpoint":"/data/myvol1"},{"Name":"myvol2","Mountpoint":"/data/myvol2"}],"Volume":null}
+```
+
+```bash
+curl -H "Content-Type: application/json" -XPOST -d '{}' --unix-socket /var/run/docker/plugins/45e00a7ce6185d6e365904c8bcf62eb724b1fe307e0d4e7ecc9f6c1eb7bcdb70/plugin.sock http:/NetworkDriver.GetCapabilities
+
+{"Scope":"local"}
+```
+When using curl 7.5 and above, the URL should be of the form `http://hostname/APICall`; ex `http://localhost/VolumeDriver.List`
```
Comment (thaJeztah) [inline docs/extend/index.md]: can you change `ex` to `for example` (or `e.g., `)?

Not sure how to put it, and if we need to mention; the "hostname" is just to make it a valid URL, but you can use any hostname you like

Comment (anusha-ragunathan) [inline docs/extend/index.md]: Updated.

**Review by thaJeztah — APPROVED** [review]
LGTM

**Pull request merged**